File: backend/app/Models/Stock.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Stock extends Model
{
    use HasFactory;

    protected $fillable = ['product_id', 'fabric_type', 'color', 'quantity_meter', 'critical_level'];

    protected $casts = [
        'quantity_meter' => 'decimal:2',
        'critical_level' => 'decimal:2',
    ];

    protected $appends = ['is_critical'];

    public function product()
    {
        return $this->belongsTo(Product::class);
    }

    public function getIsCriticalAttribute(): bool
    {
        return $this->quantity_meter < $this->critical_level;
    }
}

File: backend-dotnet/Models/Stock.cs
namespace TexTrack.Api.Models;

public class Stock
{
    public long Id { get; set; }
    public long ProductId { get; set; }
    public string FabricType { get; set; } = string.Empty;
    public string Color { get; set; } = string.Empty;
    public decimal QuantityMeter { get; set; }
    public decimal CriticalLevel { get; set; } = 100;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Product? Product { get; set; }

    public bool IsCritical => QuantityMeter < CriticalLevel;
}
